app/src/main/java/cz/muni/fi/pv256/movio2/uco_410034/MainActivity.java

```diff
@@ -0,0 +1,268 @@
+import butterknife.BindString;
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import cz.muni.fi.pv256.movio2.uco_410034.Model.Movie;
+import cz.muni.fi.pv256.movio2.uco_410034.Model.MovieCategory;
+import cz.muni.fi.pv256.movio2.uco_410034.MovieDetail.MovieDetailFragment;
+import cz.muni.fi.pv256.movio2.uco_410034.MovieList.MovieListFragment;
+import cz.muni.fi.pv256.movio2.uco_410034.MovieList.MovieSelectedListener;
+
+public class MainActivity extends AppCompatActivity implements MovieSelectedListener{
+
+    @BindView(R.id.rootLayout) DrawerLayout mDrawerLayout;
+    @Nullable @BindView(R.id.contentLayout) FrameLayout mContentLayout;
+    @Nullable @BindView(R.id.tabletContentLayout) ConstraintLayout mTabletContentLayout;
+    @Nullable @BindView(R.id.leftContentLayout) FrameLayout mLeftContentLayout;
+    @Nullable @BindView(R.id.rightContentLayout) FrameLayout mRightContentLayout;
+    @BindView(R.id.emptyView) ViewStub mEmptyView;
+
+    @BindString(R.string.fragment_movie_list_tag) String fragmentMovieListTag;
+    @BindString(R.string.fragment_movie_detail_tag) String fragmentMovieDetailTag;
+    @BindString(R.string.shared_pref_name) String sharedPrefName;
+    @BindString(R.string.shared_pref_style_key) String sharedPrefStyleKey;
+    @BindString(R.string.bundle_movie_categories_key) String mBundleMovieCategoriesKey;
+    @BindString(R.string.bundle_movie_key) String mBundleMovieKey;
+    @BindBool(R.bool.isTablet) boolean mIsTablet;
+    @BindString(R.string.empty_list_no_connection) String mEmptyListNoConnection;
+    @BindString(R.string.empty_list_no_data) String mEmptyListNoData;
+
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    private SparseArray<MovieCategory> mMovieCategories = new SparseArray<>(2);
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        SharedPreferences prefs = this.getSharedPreferences(getString(R.string.shared_pref_name), Context.MODE_PRIVATE);
+        setTheme(prefs.getInt(getString(R.string.shared_pref_style_key), R.style.AppTheme_1));
+        setContentView(R.layout.activity_main);
+        ButterKnife.bind(this);
+
+        if (savedInstanceState != null) {
+            mMovieCategories = savedInstanceState.getSparseParcelableArray(mBundleMovieCategoriesKey);
+        }
+        else {
+            fillDummyData();
+        }
+
+        setUpDrawer();
+
+        if(isDataEmpty()) {
+            setUpEmptyView();
+        }
+        else {
+            setUpFragments(savedInstanceState);
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        if(!mIsTablet && getSupportFragmentManager().findFragmentByTag(fragmentMovieDetailTag) != null) {
+            Bundle bundle = new Bundle();
+            bundle.putSparseParcelableArray(mBundleMovieCategoriesKey, mMovieCategories);
+            MovieListFragment movieListFragment = new MovieListFragment();
```

Comment:
tu by bolo fajn pouzit staticku metodu createFragment(Bundle arguments, Listener this) v triede MovieListFragment.